File: api/app/services/period_labels.py
"""Sport-aware period labels for play-by-play data."""

from __future__ import annotations


def period_label(period: int, league_code: str) -> str:
    """Return a display-ready period label.

    NBA:   Q1-Q4, OT, 2OT, 3OT …
    NHL:   P1-P3, OT, SO
    NCAAB: H1, H2, OT, 2OT, 3OT …
    """
    code = league_code.upper()

    if code == "NHL":
        if period <= 3:
            return f"P{period}"
        if period == 4:
            return "OT"
        return "SO"

    if code == "NCAAB":
        if period <= 2:
            return f"H{period}"
        ot_num = period - 2
        return "OT" if ot_num == 1 else f"{ot_num}OT"

    # NBA (default)
    if period <= 4:
        return f"Q{period}"
    ot_num = period - 4
    return "OT" if ot_num == 1 else f"{ot_num}OT"


def time_label(period: int, game_clock: str | None, league_code: str) -> str:
    """Combine period label + game clock: "Q4 2:35", "P3 12:00", "H2 5:15"."""
    plabel = period_label(period, league_code)
    if game_clock:
        return f"{plabel} {game_clock}"
    return plabel

File: web/src/lib/utils/periodLabels.ts
/**
 * Sport-aware period labels for play-by-play and game flow display.
 *
 * NBA:   Q1-Q4, OT, 2OT, 3OT ...
 * NCAAB: H1, H2, OT, 2OT, 3OT ...
 * NHL:   P1-P3, OT, SO
 */

export function formatPeriodLabel(period: number, leagueCode: string): string {
  const code = leagueCode.toUpperCase();

  if (code === "NHL") {
    if (period <= 3) return `P${period}`;
    if (period === 4) return "OT";
    return "SO";
  }

  if (code === "NCAAB") {
    if (period <= 2) return `H${period}`;
    const ot = period - 2;
    return ot === 1 ? "OT" : `${ot}OT`;
  }

  // NBA (default)
  if (period <= 4) return `Q${period}`;
  const ot = period - 4;
  return ot === 1 ? "OT" : `${ot}OT`;
}

/**
 * Format a period range label for blocks spanning multiple periods.
 * e.g. "Q1-Q2", "H1-H2", "P2-P3"
 */
export function formatPeriodRange(
  periodStart: number,
  periodEnd: number,
  leagueCode: string,
): string {
  if (periodStart === periodEnd) {
    return formatPeriodLabel(periodStart, leagueCode);
  }
  return `${formatPeriodLabel(periodStart, leagueCode)}–${formatPeriodLabel(periodEnd, leagueCode)}`;
}
